algebird-core/src/main/scala/com/twitter/algebird/Monoid.scala

```diff
@@ -47,8 +48,14 @@ trait Monoid[@specialized(Int, Long, Float, Double) T] extends Semigroup[T] {
       None
     }
   }
-  // Override this if there is a more efficient means to implement this
-  def sum(vs: TraversableOnce[T]): T = sumOption(vs).getOrElse(zero)
+  override def sum(vs: TraversableOnce[T]): T = sumOption(vs).getOrElse(zero)
+
+  /**
+   * These are from algebra.Monoid
+   */
+  override def additive: AMonoid[T] = this
+  override def empty: T = zero
+  override def combineAll(t: TraversableOnce[T]): T = sum(t)
```

Comment:
Right, this is a distinction that was important to us over in Spire-land, and which @johnynek agreed wouldn't be too difficult to encode here. I think the strategy of having Algebird's instances extend both the *generic* and *additive* traits is an elegant solution that preserves the previous semantics while aiding interop.